libpromises/eval_context.c

```diff
@@ -1443,6 +1450,8 @@ static bool EvalContextClassPut(EvalContext *ctx, const char *ns, const char *na
         return false;
     }
 
+    StringSetAdd(ctx->all_classes, SafeStringDuplicate(name));
```

Comment:
In case of masterfiles we still do 75% of the appends in this case, only thing we would optimise it to keep that 50kb map in ram for a bit shorter.